python/ray/_private/runtime_env/image_uri.py

```diff
@@ -109,28 +120,36 @@ def _modify_container_context_impl(
         container_command.append(os.getcwd())
     container_command.append("--cap-add=AUDIT_WRITE")
 
-    # mount ray package site path
-    host_site_packages_path = get_ray_site_packages_path()
-
-    # If the user specifies a `py_executable` in the container
-    # and it starts with the ${PYENV_ROOT} environment variable (indicating a PYENV-managed executable),
-    # we define `redirected_pyenv_folder` as `ray/.pyenv`.
-    # This ensures that all .pyenv-related paths are redirected
-    # to avoid overwriting the container's internal PYENV environment
-    # (which defaults to `/home/admin/.pyenv`).
     redirected_pyenv_folder = None
-    if py_executable and py_executable.startswith(get_pyenv_path()):
-        redirected_pyenv_folder = "ray/.pyenv"
-
-    host_pyenv_path = get_pyenv_path()
-    container_pyenv_path = host_pyenv_path
-    if redirected_pyenv_folder:
-        container_pyenv_path = host_pyenv_path.replace(
-            ".pyenv", redirected_pyenv_folder
-        )
-        context.container[redirected_pyenv_folder] = redirected_pyenv_folder
-    container_to_host_mount_dict[container_pyenv_path] = host_pyenv_path
-    container_to_host_mount_dict[host_site_packages_path] = host_site_packages_path
+    if install_ray:
```

Comment:
Add comment to explain why the `container_install_ray` is incompetable  with `py_executable`